File: arrays/containerWithMostWater.py
""" Container with most water - Leetcode """
def maxArea(self, height):
    """
    :type height: List[int]
    :rtype: int
    """
    i, j = 0, len(height) - 1
    area, maximum = 0, 0
        
    while i <= j:
        if height[i] > height[j]:
            area = height[j] * (j-i)
            j -= 1
        else:
            area = height[i] * (j-i)
            i += 1
        if area > maximum:
            maximum = area
    return maximum  

File: two pointers/containerWithMostWater.java
// Leetcode - 11 - Container with most water
public int maxArea(int[] height) {
    int i = 0;
    int j = height.length - 1;
    int result = Integer.MIN_VALUE;
    int product;
    while(i <= j) {
        if(height[i] <= height[j]) {
            product = (j-i) * height[i];
            i++;
        } else {
            product = (j-i) * height[j];
            j--;
        }
        result = Math.max(result, product);
    }
    return result;
}